python/ray/tests/test_autoscaler_yaml.py

```diff
@@ -396,6 +397,126 @@ def testAzureKeyPair(self):
             assert node_type["node_config"]["azure_arm_parameters"][
                 "publicKey"] == "PUBLICKEY"
 
+    @pytest.mark.skipif(
+        sys.platform.startswith("win"), reason="Fails on Windows.")
+    def testGCPSubnets(self):
+        """Validates gcp _configure_subnet logic.
+
+        Checks that _configure_subnet fills default networkInterfaces data for
+        each node type that doesn't specify networkInterfaces.
+
+        Checks that _list_subnets is not called if all node types specify
+        networkInterfaces.
+        """
+        path = os.path.join(RAY_PATH, "autoscaler", "gcp", "example-full.yaml")
+        config = yaml.safe_load(open(path).read())
+
+        config_subnets_configured = copy.deepcopy(config)
+        config_subnets_worker_configured = copy.deepcopy(config)
+        config_subnets_head_configured = copy.deepcopy(config)
+        config_subnets_no_type_configured = copy.deepcopy(config)
+
+        config_subnets_configured["available_node_types"]["ray_head_default"][
+            "node_config"]["networkInterfaces"] = "mock_interfaces"
+        config_subnets_configured["available_node_types"]["ray_worker_small"][
+            "node_config"]["networkInterfaces"] = "mock_interfaces"
+
+        config_subnets_worker_configured["available_node_types"][
+            "ray_worker_small"]["node_config"][
+                "networkInterfaces"] = "mock_interfaces"
+
+        config_subnets_head_configured["available_node_types"][
+            "ray_head_default"]["node_config"][
+                "networkInterfaces"] = "mock_interfaces"
+
+        assert "networkInterfaces" not in config_subnets_no_type_configured[
+            "available_node_types"]["ray_head_default"]["node_config"]
+        assert "networkInterfaces" not in config_subnets_no_type_configured[
+            "available_node_types"]["ray_worker_small"]["node_config"]
+
+        # Configure subnets modifies configs in place so we need to copy
+        # the configs for comparision after passing into the method.
+        config_subnets_configured_post = copy.deepcopy(
+            config_subnets_configured)
+        config_subnets_worker_configured_post = copy.deepcopy(
+            config_subnets_worker_configured)
+        config_subnets_head_configured_post = copy.deepcopy(
+            config_subnets_head_configured)
+        config_subnets_no_type_configured_post = copy.deepcopy(
+            config_subnets_no_type_configured)
+
+        # Track number of times list_subnets has been called.
+        list_subnets_counter = 0
+
+        def mock_list_subnets(*args):
+            nonlocal list_subnets_counter
+            list_subnets_counter += 1
+            return [{"selfLink": "link"}]
+
+        with patch("ray.autoscaler._private.gcp.config._list_subnets",
```

Comment:
this line leads to the error below in the CI but not locally. Anyone have an idea what's going on?
```
thing = <module 'ray.autoscaler._private.gcp' from '/ray/python/ray/autoscaler/_private/gcp/__init__.py'>
--
  | comp = 'config', import_path = 'ray.autoscaler._private.gcp.config'
  |  
  | def _dot_lookup(thing, comp, import_path):
  | try:
  | >           return getattr(thing, comp)
  | E           AttributeError: module 'ray.autoscaler._private.gcp' has no attribute 'config'

```